addon/lib/StudyTelemetryCollector.jsm

```diff
@@ -135,15 +136,18 @@ class StudyTelemetryCollector {
 
       const shieldPingPayload = StudyTelemetryCollector.createShieldPingPayload(shieldPingAttributes);
 
+      // Add additional add-on metadata to the payload since pioneer-utils doesn't seem to do include the same metadata as shield-utils
+      shieldPingPayload.pioneerAddonMetadata = this.pioneerUtils.pioneerAddonMetadata;
```

Comment:
this all seems like a really convoluted/hacky way of sending through *metadata*. It sounds like pioneer-utils needs to be modified to do what you need. I think modifying pioneer-utils might be a better move. It also looks like this made your schema way more complicated than it should be.

Reply:
Yes, it would be great if pioneer-utils already sent this metadata similarly to how shield-utils does. Do you want to add it today or should we aim to have it added for upcoming versions of the addon?

Reply:
I will investigate what shield utils does but will likely not be able to land this change before monday.

Reply:
Shield utils wraps the payload with this kind of metadata here https://github.com/mozilla/shield-studies-addon-utils/blob/master/src/StudyUtils.in.jsm#L353-L366 and it is very useful to be able to weed out pings associated with outdated versions of the addon, certain branches, testing pings etc.

Reply:
ah I see. this is probably not something we would do with pioneer-utils, as the pipeline is very different for pioneer pings.